### PR TITLE
Added IDs, guid gen, tests

### PR DESCRIPTION
**Pull request by IlyaMoskva**
Solved issue #46.
IDs are required to simplify implementation of save/load graph functionality.
GUID is a common class is a random String provider.
Tested new functionality and coverage increased.

### CODE REVIEW

**Review by stepanvalyavskiy — CHANGES_REQUESTED** [review]
I have no permissions to update a branch. Could you please update it, while it contains changes from PR #45

**Comment by IlyaMoskva** [comment]
@stepanvalyavskiy Rebased. Please, check again

File: src/main/com/cqfn/degitx/simulator/randomengine/GUID.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.cqfn.degitx.simulator.randomengine
+
+import java.util.*
+import kotlin.streams.asSequence
+
+const val longSource = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+
+const val GUIDLength = 8L
+
+class GUID {
```
Comment (stepanvalyavskiy) [inline src/main/com/cqfn/degitx/simulator/randomengine/GUID.kt:10]: please add comments

File: src/test/com/cqfn/degitx/simulator/data/GraphDataTest.kt [hunk 2]
```diff
@@ -19,7 +25,7 @@ class GraphDataTest {
 
     @Test
     fun witeStaticGraph() {
-        graph.generate(GraphGenerator.Settings())
+        // graph.generate(GraphGenerator.Settings())
```
Comment (stepanvalyavskiy) [inline src/test/com/cqfn/degitx/simulator/data/GraphDataTest.kt:28]: please remove commented out code.

File: src/test/com/cqfn/degitx/simulator/RequestPathTest.kt [hunk 3]
```diff
@@ -1,21 +1,17 @@
...
 
-    @BeforeTest
-    fun setUp() {
-        graph.generate(GraphGenerator.Settings())
-    }
-
     @Test
     fun requestGoesToTarget() {
+        graph.generate(GraphGenerator.Settings())
+
```
Comment (stepanvalyavskiy) [inline src/test/com/cqfn/degitx/simulator/RequestPathTest.kt:14]: It could be moved to @BeforeSuite fun as in other test classes

**Review by stepanvalyavskiy — CHANGES_REQUESTED** [review]
Hi, @IlyaMoskva, please check comments.

File: src/test/com/cqfn/degitx/simulator/RequestPathTest.kt (hunk 3 above)
Comment (IlyaMoskva) [reply src/test/com/cqfn/degitx/simulator/RequestPathTest.kt:14]: It was so, but does not work properly - test failed. I don't found out the reason for now, so this is a temporary solution.

File: src/test/com/cqfn/degitx/simulator/data/GraphDataTest.kt (hunk 2 above)
Comment (IlyaMoskva) [reply src/test/com/cqfn/degitx/simulator/data/GraphDataTest.kt:28]: Sure, done.

File: src/main/com/cqfn/degitx/simulator/randomengine/GUID.kt (hunk 1 above)
Comment (IlyaMoskva) [reply src/main/com/cqfn/degitx/simulator/randomengine/GUID.kt:20]: @stepanvalyavskiy ,thank for catching that missing. Fixed.

**Review by stepanvalyavskiy — APPROVED** [review]
@IlyaMoskva, thanks

**Pull request merged**